src/defaultsettings.cpp

```diff
@@ -104,7 +104,15 @@ void set_default_settings()
 	// Client
 	settings->setDefault("address", "");
 	settings->setDefault("enable_sound", "true");
+#if defined(__linux__) && !defined (__ANDROID__)
+	// On Linux+X11 (not Linux+Wayland or Linux+XWayland), I've encountered a bug
+	// where fake mouse events were generated from touch events if in relative
+	// mouse, resulting in the touchscreen controls being instantly disabled again
+	// and thus making them unusable. Therefore, I've disabled automatic switching.
+	settings->setDefault("touch_controls", "false");
```

Comment:
Obviously suboptimal. I still don't even know whether that bug is a general one or specific to my system.

Reply:
I would go with `bool_to_cstr(has_touch)` here (for the workaround)